src/util.js

```diff
@@ -1,175 +1,13 @@
   return array.indexOf(item) !== -1;
 };
 
-util.uniq = function(data, field) {
-  var map = {}, count = 0, i, k;
-  for (i = 0; i < data.length; ++i) {
-    k = data[i][field];
-    if (!map[k]) {
-      map[k] = 1;
-      count += 1;
-    }
-  }
-  return count;
-};
-
-var isNumber = function(n) {
-  return !isNaN(parseFloat(n)) && isFinite(n);
-};
-
-// try parsing to number
-util.numbers = function(values) {
-  var nums = [];
-  for (var i = 0; i < values.length; i++) {
-    if (isNumber(values[i])) {
-      nums.push(+values[i]);
-    }
-  }
-  return nums;
-};
-
-// try to parse as date
-util.dates = function(values) {
-  var dates = [];
-  for (var i = 0; i < values.length; i++) {
-    var date = Date.parse(values[i]);
-    if (!isNaN(date)) {
-      dates.push(new Date(date));
-    }
-  }
-  return dates;
-};
-
-util.median = function(values) {
-  values.sort(function(a, b) {return a - b;});
-  var half = Math.floor(values.length/2);
-  if (values.length % 2) {
-    return values[half];
-  } else {
-    return (values[half-1] + values[half]) / 2.0;
-  }
-};
-
-util.mean = function(values) {
-  return values.reduce(function(v, r) {return v + r;}, 0) / values.length;
-};
-
-util.variance = function(values) {
-  var avg = util.mean(values);
-  var diffs = [];
-  for (var i = 0; i < values.length; i++) {
-    diffs.push(Math.pow((values[i] - avg), 2));
-  }
-  return util.mean(diffs);
-};
-
-util.stablesort = function(array, sortBy, keyFn) {
-  var indices = {};
-
-  array.forEach(function(v, i) {
-    indices[keyFn(v)] = i;
-  });
-
-  array.sort(function(a, b) {
-    var sa = sortBy(a),
-      sb = sortBy(b);
-
-    return sa<sb ? -1 : sa>sb ? 1 : (indices[keyFn(a)] - indices[keyFn(b)]);
-  });
-  return array;
-};
-
-util.stdev = function(values) {
-  return Math.sqrt(util.variance(values));
-};
-
-util.skew = function(values) {
-  var avg = util.mean(values),
-    med = util.median(values),
-    std = util.stdev(values);
-  return 1.0 * (avg - med) / std;
-};
-
-// parses a string to date or number
-util.parse = function(value) {
-  if (isNumber(value)) {
-    return +value;
-  }
-
-  var date = Date.parse(value);
-  if (!isNaN(date)) {
-    return (new Date(date));
-  }
-  return value;
-};
-
-util.minmax = function(data) {
-  var stats = {min: +Infinity, max: -Infinity};
-  for (var i = 0; i < data.length; ++i) {
-    var v = data[i];
-    if (v !== null) {
-      if (v > stats.max || stats.max === -Infinity) stats.max = v;
-      if (v < stats.min || stats.min === +Infinity) stats.min = v;
-    }
-  }
-  return stats;
-};
-
-util.duplicate = function(obj) {
-  return JSON.parse(JSON.stringify(obj));
-};
-
-util.isObject = function(obj) {
-  return obj === Object(obj);
-};
-
-util.isArray = Array.isArray || function(obj) {
-  return toString.call(obj) == '[object Array]';
-};
-
-util.array = function(x) {
-  return x ? (util.isArray(x) ? x : [x]) : [];
-};
-
 util.forEach = function(obj, f, thisArg) {
```

Comment:
The array version should use `for(var i; i<n;i++)` for performance reasons. Anyway, I'm okay with keeping this here for now but let's not push it into datalib.